tester_app/client/app/bundles/HelloWorld/containers/HelloWorldContainer.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { PropTypes } from 'react';
+import HelloWorld from '../components/HelloWorld';
+
+// Simple example of a React component
+export default class HelloWorldContainer extends React.Component {
+  static displayName = 'HelloWorldContainer';
+
+  static propTypes = {
+    initialName: PropTypes.string.isRequired, // this is passed from the Rails view
+  }
+
+  state = {name: this.props.initialName} // how to set initial state in es2015 class syntax
+
+  updateName(name) {
+    this.setState({name: name});
+  }
+
+  render() {
+    const updateName = ::this.updateName;
+    const name = this.state.name;
```

Comment:
Usually there are more than one property on `props` / `state` objects, so destructuring makes code, where those properties are used, more readable: less noise, shorter lines (I follow 80 chars limit rule).

```js
{
  this.props.isLoading &&
  <div>{this.props.firstName} {this.props.lastName}</div>
}

/* VS */

const { firstName, lastName, isLoading } = this.props;

{isLoading && <div>{firstName} {lastName}</div>}
```

So I just got used to it.

@jbhatab @justin808 @robwise